CustomLookupTest/test.cs

```diff
@@ -0,0 +1,286 @@
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Tests;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Linq.Expressions;
+using System.Reflection.Metadata;
+using System.Threading;
+using System.Timers;
+using System.Linq;
+using System.Runtime.ExceptionServices;
+/// <summary>
+/// This is my attempt at a unit test. I took information from VSE documentation and the team's current SkillsetsTest.cs
+/// </summary>
+namespace Test
+{
+
+    [TestClass]
+    public class CustomLookupTests
```

Comment:
It seems like your tests are a little more than unit tests: there should be no need to call your function using a web request. By doing so, you're testing the Azure functions infrastructure rather than just your code. I'd re3factor to just pass data into your function directly.

Reply:
I am still unsure how I am supposed to pass data into my function directly without using Moq or some "substitute" client. If possible, could you explain how I should go about running my tests this way?

Reply:
You should be able to just call directly into the function. 
EG:
CustomEntitySearch.Run(TestData.GeneratePayloadRequest(), new TestLogger(), new ExecutionContext(~~,~~,~))

---
In reply to: [304646902](https://github.com/Azure-Samples/azure-search-power-skills/pull/1#discussion_r304646902) [](ancestors = 304646902)

Reply:
Add a project reference to the main project from your test project, and call `CustomEntitySearch.Run` directly instead of doing a `PostRequest`.

Reply:
You may also want to add a helper method to your test project that builds the `HttpRequest` from text and words, which should remove a lot of the repetition in those tests.